### PR TITLE
Refactor ledger-tool moving some of functions to a separate module

### PR DESCRIPTION
**Pull request by dmakarov**
#### Problem

ledger-tool main.rs has overgrown. Some of its functions can be reused by other commands.

#### Summary of Changes

This PR factors out several functions from main.rs to a separate module for reuse in run command added in #31346

### CODE REVIEW

**Comment by dmakarov** [comment]
Thank you. Will wait for Steve to approve before merging.

**Comment by dmakarov** [comment]
@steviez unless you’re still reviewing I’d like to merge this.

**Comment by steviez** [comment]
> @steviez unless you’re still reviewing I’d like to merge this.

Yes, I'd still like to review this. I'll try to knock it out shortly - I was tied up yesterday hunting down another bug that we need for master to be compatible with clusters.

File: ledger-tool/src/main.rs
Comment (dmakarov) [reply ledger-tool/src/main.rs:3]: How about `blockstore.rs`? Does this work?

Comment (CriesofCarrots) [reply ledger-tool/src/main.rs:3]: Maybe `open_blockstore.rs` to disambiguate from `solana_ledger::blockstore`?

Comment (steviez) [reply ledger-tool/src/main.rs:3]: There is specifically non-blockstore stuff getting moved, such as the CLI parsing function `get_accounts_db_config()`

Comment (CriesofCarrots) [reply ledger-tool/src/main.rs:3]: > There is specifically non-blockstore stuff getting moved, such as the CLI parsing function `get_accounts_db_config()`

Yeah, maybe accountsDb-related stuff should be in a separate module.

Comment (dmakarov) [reply ledger-tool/src/main.rs:3]: ok to move those functions to  accounts_db.rs then?

Comment (dmakarov) [reply ledger-tool/src/main.rs:3]: yes, that was my reason for putting them all in utils.rs originally. How about for now, I rename `utils.rs` to `ledger_utils.rs` and then in the future when more refactoring is done I'll split in smaller modules?   My main goal is to get the new command into ledger-tool at the moment, see #31346 if interested.

Comment (CriesofCarrots) [reply ledger-tool/src/main.rs:3]: > There is another function, `hardforks_of()`, that parses `matches: &ArgMatches<'_>`. Thoughts on putting this one and `get_accounts_db_config` in something like `cli_utils`?

This wfm. It would actually be nice to break out the ledger-tool cli and arg parsing entirely from the command processing one of these days (we have precedent for `cli.rs` and `args.rs/arg_parser.rs`)

> Also, `load_bank_forks()` operates on both blockstore and accounts db, and is beyond just opening blockstore. I know @CriesofCarrots mentioned not wanting very generic module name `utils`, but would you be happy with `ledger_utils` ?

I don't love it, but haven't thought of anything better, assuming you're not game for a separate `load_bank_forks.rs`

Comment (steviez) [reply ledger-tool/src/main.rs:3]: > This wfm. It would actually be nice to break out the ledger-tool cli and arg parsing entirely from the command processing one of these days (we have precedent for cli.rs and args.rs/arg_parser.rs)

Yeah, I was actually thinking the same thing. I have a plane ride today, but I can take a swing at it then. Definitely no need for @dmakarov to bite that off in this PR IMO

> I don't love it, but haven't thought of anything better, assuming you're not game for a separate load_bank_forks.rs

That's fair. As I see it, the groupings of function we have are:
- 2 arg parser functions: `hardforks_of()` and `get_accounts_db_config()`
- 3 blockstore related functions: `get_shred_storage_type()`, `open_blockstore()`, `open_blockstore_with_temporary_primary_access()`
- `load_bank_forks()`
- `open_genesis_config_by()`

I think the arg parsing functions in one module for this PR make sense, to be later updated by more expansive cli/arg overhaul.

I was then trying to come up with a catch-all for the other 5 functions. I like using specific module names too, but if we get too specific, I feel like new functions may not quite fit and we might end up with a handful of modules with a single function in them (which I guess isn't technically a bad thing)

Comment (dmakarov) [reply ledger-tool/src/main.rs:3]: So what do we settle on for this PR?

Comment (CriesofCarrots) [reply ledger-tool/src/main.rs:3]: > I think the arg parsing functions in one module for this PR make sense, to be later updated by more expansive cli/arg overhaul.
> 
> I was then trying to come up with a catch-all for the other 5 functions.

I definitely don't think single-function modules are bad :) But since we know we need a lot more reorg in solana-ledger-tool, I'm on board with doing this ^ for now, ie. arg-parsing in one module, other methods in `ledger_utils.rs`. At least ledger_utils tells me that the stuff in it is related to the functionality in `solana-ledger`.

Comment (dmakarov) [reply ledger-tool/src/main.rs]: I moved arg parsing functions to a new module `args.rs` and renamed the `utils.rs` module to `ledger_utils.rs`.

**Review by CriesofCarrots — APPROVED** [review]
Thanks for the back-and-forth! Lgtm with @steviez 's ✅

**Comment by dmakarov** [comment]
@steviez are we good to merge?

**Pull request merged**
